javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollector.java

```diff
@@ -201,6 +203,73 @@ private void collectHelperClasses(
     return references;
   }
 
+  public void prune() {
+    Set<Reference> helperClassesWithLibrarySuperType = getHelperClassesWithLibrarySuperType();
+
+    Set<Reference> needToKeepFieldsAndMethods = new HashSet<>();
+    for (Reference reference : helperClassesWithLibrarySuperType) {
+      addSuperClasses(reference.getClassName(), needToKeepFieldsAndMethods);
+    }
```

Comment:
Can you add a comment explaining why this code is needed? E.g `A extends B implements LibraryInterface` situation where actual `LibraryInterface` implementations is in B